src/components/Home/CTAButton.tsx

```diff
@@ -1,27 +1,107 @@
 import { useTranslation } from "next-i18next";
 import { trackUmamiEvent } from "../../lib/umami";
-import Button from "../Button";
+import Button, { ButtonType } from "../Button";
+import { AletheiaModal } from "../Modal/AletheiaModal.style";
+import colors from "../../styles/colors";
+import { useState } from "react";
+import AletheiaButton from "../Button";
 
-const CTAButton = ({ type }) => {
+interface CTAButtonProps {
+    type?: string;
+    isLoggedIn?: boolean;
+    smallDevice?: boolean;
+    mediumDevice?: boolean;
+}
+
+const CTAButton: React.FC<CTAButtonProps> = ({
+    isLoggedIn,
+    mediumDevice = false,
+}) => {
     const { t } = useTranslation();
+    const [isModalVisible, setIsModalVisible] = useState(false);
+
+    const handleHideModal = () => {
+        setIsModalVisible(false);
+    };
+
+    const handleClick = () => {
+        trackUmamiEvent("cta-registration-button", "registration");
```

Comment:
Separate the Umami tracking from the CTA registration logic, as they represent different metrics and should be tracked independently